addons/hide-ads/userscript.js

```diff
@@ -0,0 +1,176 @@
+    const repeatedCharactersRegex = /(.)\1{20,}/;
+
+    const meaninglessRegex = /^[\W\s]*$/;
+
+    if (repeatedCharactersRegex.test(content)) {
+      return true;
+    }
+
+    if (meaninglessRegex.test(content)) {
+      return true;
+    }
+
+    return false;
+  }
+
+  /* 
+  PROFILES
+  top-level-reply
+
+  comment
+  info
+  content.innerText
+
+  replies
+  reply
+  comment
+  content.innerText
+
+  PROJECTS and STUDIOS
+  comment-container
+
+  comment
+  comment-body
+  comment-bubble
+  comment-content
+  emoji-text.innerText
+
+  replies
+  comment
+  comment-body
+  comment-bubble
+  comment-content.lastChild.innerText (in replies there are two emoji-texts)
+  */
+
+  let idToContent = {};
+
+  function handleComment(element, blockType) {
+    if (addon.settings.get("method") === "hide") {
+      element.style.display = "none";
+    } else {
+      const comment = element.classList.contains("comment") ? element : element.querySelector(".comment");
+      comment.classList.add("contains-advertising");
+
+      let commentContent;
+      let profileReplyButton;
+
+      if (url.includes(settings["profiles"])) {
+        commentContent = element.querySelector(".info>.content");
+        profileReplyButton = commentContent.nextElementSibling.lastElementChild;
+        console.log(profileReplyButton);
+      } else {
+        commentContent = element.querySelector(".comment-content");
+      }
+
+      const key = comment.getAttribute("id");
+      const value = commentContent.innerHTML;
+
+      idToContent[key] = value;
+
+      commentContent.innerHTML =
+        blockType === "content" ? advertisingContent : blockType === "user" ? advertisingUser : advertisingSpam;
+      commentContent.classList.add("advertising");
+      commentContent.style.cursor = "pointer";
+
+      if (profileReplyButton.classList.contains("reply")) {
+        profileReplyButton.style.display = "";
+      }
+
+      commentContent.addEventListener("click", () => {
+        if (commentContent.classList.contains("advertising")) {
+          commentContent.innerHTML = idToContent[commentContent.closest(".comment").getAttribute("id")];
+          commentContent.classList.remove("advertising");
+          comment.classList.remove("contains-advertising");
+
+          if (profileReplyButton.classList.contains("reply")) {
+            profileReplyButton.style.display = "inline";
+          }
+        } else {
+          commentContent.innerHTML =
+            blockType === "content" ? advertisingContent : blockType === "user" ? advertisingUser : advertisingSpam;
+          commentContent.classList.add("advertising");
+          comment.classList.add("contains-advertising");
+
+          if (profileReplyButton.classList.contains("reply")) {
+            profileReplyButton.style.display = "";
+          }
+        }
+      });
+    }
+  }
+
+  //dsis = does string include string
+  function dsis(comment, string) {
```

Comment:
`stringIncludesString` or `includesSubstring` is a better function name than `dsis`

Reply:
Not a big deal but ok